### DISCUSSION TITLE
Exposing the $magic method externally is like the store() function.

### DISCUSSION THREAD

**Discussion by goSeeFuture** [alpinejs/alpine#3894] [3. Feature ideas]
Sometimes we need to access the $magic function directly in JavaScript, for example, when using the [alpinejs-notifications](https://github.com/markmead/alpinejs-notifications) plugin, a common scenario is to give a notification after a successful click on the submit form, then we need to call the notification function directly in JavaScript, but I looked up the `alpinejs` docs and the source code, and eventually I found that the `magics` object isn't exposed to the public, so maybe we can use it like the `store()` function.

```js
export function magic(name, callback) {
    // expose magic function for javascript call
    if (callback === undefined) {
        return magics[name];
    }
    magics[name] = callback
}
```
This allows us to use notifications in this way
```js
Alpine.magic('notify')('Hello there, I am a notification!', {
    wrapperId: 'notificationWrapper',
    templateId: 'notificationAlert'
});
```

**Comment by ekwoka** [comment]
This won't make a ton of sense.

Since virtually all of the core magics (And many others) are element bound.

What these should do instead, like some of the core plugins do, is add a way to call it on the Alpine (or other accessible means).

With that in mind, this becomes more of a Plugin dev issue, as all the Alpine provided magics are element bound or available on `Alpine`

**Reply by goSeeFuture** [reply]
Thanks for your reply，I get it now.
